app/src/main/java/com/aconno/sensorics/ui/settings/publishers/selectpublish/MqttPublisherActivity.kt

```diff
@@ -75,10 +76,15 @@ class MqttPublisherActivity : BaseActivity() {
             }
         }
 
-        override fun onConnectionFail() {
+        override fun onConnectionFail(exception: Throwable?) {
             GlobalScope.launch(Dispatchers.Main) {
                 progressbar.visibility = View.INVISIBLE
                 isTestingAlreadyRunning = false
+                if(exception is IllegalArgumentException) {
```

Comment:
Check if exception contains the entered URL not if its an illegal argument exception